composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/AddSpotView.kt

```diff
@@ -310,12 +281,32 @@ fun AddSpotView(onBackClick: () -> Unit, onSpotAdded: () -> Unit, addSpotViewMod
 
             Button(
                 onClick = {
-                    val address = searchedSuggestion.value?.name ?: "dev spot"
-                    val spot = Spot("999", "dev3", address, currentLocation.longitude, currentLocation.latitude)
-                    scope.launch {
-                        addSpotViewModel.createSpot(spot)
+                    val createdSpotNumbers = parseNumberRange(spotNumbers)
+
+                    if (selectedBuildingType == "House") {
```

Comment:
Mine is showing the `dev spot x`, why is the searchSuggestion `null` sometimes?

I also think this if condition is unnecessary, if `parseNumberRange(spotNumbers)` can just return the right list of spots (including the single case) then we can reduce logic here.

Reply:
I updated this to keep the spot number list as part of the UI regardless of the type of building in the latest change. I also fixed the issue with the search suggestion so it correctly propogates the selected address with the spot number.